rust/src/buffer.rs

```diff
@@ -89,12 +89,6 @@ impl<T> Iterator for BufferIterator<T> where T: Copy {
     }
 }
 
-impl<T> Drop for Buffer<T> {
```

Comment:
master doesn't compile as this is also at https://github.com/apache/arrow/pull/1818/files/5449cc074163ec160e366ad5be84422cfdea7ccb#diff-efb40dbeb7a3acab561ac659c2cfee74R67

Reply:
@maxim-lian Would you be able to fix this in a PR?

Reply:
Yes - this the fix!